source/agora/utils/InetUtils.d

```diff
@@ -174,6 +186,135 @@ version (Windows)
     }
 }
 
+    /***************************************************************************
+        Converts a compressed IPv6 representation to uncompressed representation
+
+        For example:
+            ::5:6:7:8      => 0:0:0:0:5:6:7:8
+            1:2:3::7:8     => 1:2:3:0:0:0:7:8
+            ::             => 0:0:0:0:0:0:0:0
+
+        Params:
+            ip = compressed IPv6 address
+
+        Returns:
+            uncompressed representation of IPv6 address
+
+    ***************************************************************************/
+
+    public static string convertCompressedIPv6ToUncompressed (string ip) @safe pure
+    {
+        auto parts = ip.split(":");
+        const first_empty_ind = parts.countUntil("");
+
+        // filter out all the empty parts, except the first one
+        auto res = zip(iota(parts.length), parts).filter!((tup) => tup[0] == first_empty_ind || !tup[1].empty()).
+                   map!(tup => tup[1]).array();
+
+        // fill in the empty part with the needed zeros
+        return res.map!(part => (part.empty() ? replicate(["0"],8 - res.length + 1) : [part]))
+                    .joiner.array.join(":");
```

Comment:
```suggestion
                    .joiner(':').array();
```

?

Reply:
what I would like here is to flatten embedded maps:
```D
    auto a = [["1"],["2"],["0","0"],["5"]];
    writeln(a.map!(e => e).joiner().array.join(":"));
    // should print out 1:2:0:0:5
```
```D
writeln(a.map!(e => e).joiner(":").array; 
```
returns a compile error